### DISCUSSION TITLE
Missing artists and album artists

### DISCUSSION THREAD

**Discussion by brettpim** [LMS-Community/slimserver#872] [General]
I have the soundtrack to Amadeus ripped as flac.  I have "Wolfgang Amadeus Mozart" in the 'artist' tag and the various musicians in the 'album artists' tag.  None of them are showing up in my list of artists.  Additionally when I click on one of the tracks, I do not get the "view tags" link at the end of all the information.  It also happens with other albums.  How can I diagnose and fix this problem?

**Comment by michaelherger** [comment]
Are you saying that none of the other artists (the album artists) are showing up? Or none at all? What is your setting for the Artists list in Settings/My Music? Is there anything in your server.log or scanner.log related to those files?

Would you mind sharing a sample file with me (https://www.dropbox.com/request/T3RctyzGgNg0oFDubq6a)? Feel free to strip off the audio data.

**Comment by brettpim** [comment]
I did cleared an rescanned my library yesterday.  For the "Amadeus" the artist (Mozart) does not show up but the album artists (musicians) do.  The "view tags" is available when I select a track and "Wolfgang Amadeus Mozart" is listed as the artists and I can click on his name and successfully find all the tracks/albums he is an artist on. 

My settings in Settings/My Music are:

Browse Artists: Use single, configurable list of artists

Composer, Band and Orchestra in Artists: everything is checked

There is nothing in the scanner.log nor server.log relating to this albums

I just did a search form "Bach" in the artists and he does not show up either, while the album artists on my Bach albums do.

**Reply by michaelherger** [reply]
So it's filed under "no artist"?

**Comment by brettpim** [comment]
Where do I find "No Artist"?  It is not in the list of artists under 'N'

**Comment by michaelherger** [comment]
Ok asked differently: how do you even find that album? Can you provide some screenshots? And the scanner.log.zip after a full wipe & rescan?

**Comment by brettpim** [comment]
I search for something in the track title to find the album.  I am at work now but will post pictures and scanner.log when I get home.  Thank you for your help

**Comment by brettpim** [comment]
I did the full wipe and rescan.  Here are screenshots

track from a mozart album:
![Screenshot from 2023-03-17 08-33-04](https://user-images.githubusercontent.com/8606362/225906073-9b725878-5f05-4e9c-8322-098e599a5070.png)

viewing "all tags":
![Screenshot from 2023-03-17 08-33-22](https://user-images.githubusercontent.com/8606362/225906177-702ede61-398a-4a81-ba64-27eecd24db30.png)

searching for artists "mozart":
![Screenshot from 2023-03-17 08-32-30](https://user-images.githubusercontent.com/8606362/225906232-5327d138-ca1a-40f8-a704-d7598cac8e02.png)

searching for album artists "BBC symphony":
![Screenshot from 2023-03-17 08-33-44](https://user-images.githubusercontent.com/8606362/225906293-deaa757c-c3ca-4dbd-80db-5c54389cfaed.png)

scanner.log.zip:
[scanner.log.zip](https://github.com/Logitech/slimserver/files/11001368/scanner.log.zip)

**Reply by michaelherger** [reply]
I'm a bit confused. In the initial posting you said 

>  I do not get the "view tags" link at the end of all the information.

But here you're posting a screenshot not only of that very menu item, but also of its content.

And from what I understand you have album artist which vary from track to track, but always the same track artist? Isn't this the wrong way around? Wouldn't you have one album artist (Mozart), and then the various artists playing on the tracks?

**Reply by brettpim** [reply]
Yes in my second post, I corrected myself that I did have the "view tags".  

It would be possible to tag my classical tracks the way you suggest but often the album artists are the same on an album but the composers differ between tracks so I thought album artists was more appropriate for the musicians who are consistent.  Of course this is not always true but I had to pick one way or the other.

**Reply by michaelherger** [reply]
Classical music is complicated 😀. I mean... there are "composer" tags, and "conductor". But you put everything in the album artist. IMHO that's wrong.

**Reply by brettpim** [reply]
I did not know about these tags.  Does FLAC support these tags?  I am up for retagging if there is a better system.  But regardless of my tagging system, I still do think that both Artist and Album Artists should show up on squeezebox server

**Reply by michaelherger** [reply]
I think FLAC uses Vorbis tags. See eg. https://de.wikipedia.org/wiki/Vorbis_comment

**Reply by brettpim** [reply]
thanks for this; I am looking into what is available

**Comment by brettpim** [comment]
Any ideas about diagnosing and fixing the original problem I am having: "Album Artist"s being indexed, but not "Artist"?

**Comment by michaelherger** [comment]
Not really - unless you speak SQL. In that case you could look at the debug output of the `database.sql` log setting (Settings/Advanced/Logging).
